### PR TITLE
Ensure child kernel is ready to go

### PR DESCRIPTION
**Pull request by rgbkrk**
This creates an AsyncKernelClient via the (internal) AsyncKernelManager and uses it to check on our "child" kernel being ready to go.

### CODE REVIEW

**Comment by rgbkrk** [comment]
Mmmmm TODO Bot. Fun to be back in the land of nteract. 😄

File: pick_kernel/kernel.py [hunk 1]
```diff
@@ -293,15 +332,16 @@ async def launch_kernel_with_parameters(self, stream, ident, parent, config):
         self._publish_status("idle")
 
     def parse_cell(self, cell):
-        if not cell.startswith("%%kernel.config"):
+        if not cell.startswith("%%kernel."):
```
Comment (rgbkrk) [inline pick_kernel/kernel.py:335]: I could/should break the following changes off into another PR. It was just nice to have a fake namespace while working with `pick` right now (until we have registration).

Comment (MSeal) [reply pick_kernel/kernel.py:335]: we can iterate on that over time for sure

File: pick_kernel/kernel.py [hunk 2]
```diff
@@ -151,19 +149,60 @@ async def start_kernel(self, config=None):
...
+                pass
+            else:
+                if msg["msg_type"] == "kernel_info_reply":
+                    # Now we know the kernel is (mostly) ready.
+                    # However, most kernels are not quite ready at this point to send
+                    # on more execution.
+                    #
+                    # Do we wait for a further status: idle on iopub?
+                    # Wait for idle?
+                    # Wait for particular logs from the stdout of the kernel process?
```
Comment (MSeal) [inline pick_kernel/kernel.py:186]: Hmm probably don't want to wait for logs... waiting for idle may be valid, but getting the kernel_info_reply does mean the process is alive. Might be ok to just let the downstream kernel continue processing messages from here as normal?

Comment (rgbkrk) [reply pick_kernel/kernel.py:186]: The most common issue I ran into was that shell would be ready but iopub would not, and so execution messages would happen and we'd miss some batch of iopub. This was easiest to reproduce with a restart-and-run-all command from the Jupyter UI.
